bridges/otelzap/encoder.go

```diff
@@ -55,15 +55,13 @@ func (m *objectEncoder) calculate(o *namespace) {
 }
 
 func (m *objectEncoder) AddArray(key string, v zapcore.ArrayMarshaler) error {
-	// TODO: Use arrayEncoder from a pool.
 	arr := &arrayEncoder{}
 	err := v.MarshalLogArray(arr)
 	m.cur.attrs = append(m.cur.attrs, log.Slice(key, arr.elems...))
 	return err
 }
 
 func (m *objectEncoder) AddObject(k string, v zapcore.ObjectMarshaler) error {
-	// TODO: Use objectEncoder from a pool.
 	newobj := newObjectEncoder(2)
```

Comment:
It is always `2`. Maybe we can move the magic value inside `newObjectEncoder` (and just have `newobj := newObjectEncoder()` and also explain why it is 2 and not any different value.

This can be handled in a separate PR.

Reply:
we don't do that because here we initialize it based on length of the field

```
func convertField(fields []zapcore.Field) (context.Context, []log.KeyValue) {
	var ctx context.Context
	enc := newObjectEncoder(len(fields))
	.... 
}
```